src/drivers/nuopc/cime_flds/esmFldsExchange.F90

```diff
@@ -1109,62 +1112,79 @@ subroutine esmFldsExchange(gcomp, phase, rc)
+       if (phase == 'advertise') then
           ! 'fractional ice coverage wrt ocean for each thickness category '
           call addfld(fldListFr(compice)%flds, 'Si_ifrac_n')
+          call addfld(fldListTo(compocn)%flds, 'Si_ifrac_n')
+
           ! net shortwave radiation penetrating into ocean for each thickness category
           call addfld(fldListFr(compice)%flds, 'Fioi_swpen_ifrac_n')
-          ! 'fractional atmosphere coverage wrt ocean'
+          call addfld(fldListTo(compocn)%flds, 'Fioi_swpen_ifrac_n')
+
+          ! 'fractional atmosphere coverage wrt ocean' (computed in med_phases_prep_ocn)
           call addfld(fldListTo(compocn)%flds, 'Sf_afrac')
-          ! 'net shortwave radiation times atmosphere fraction'
-          call addfld(fldListTo(compocn)%flds, 'Foxx_swnet_afracr')
-          ! 'fractional atmosphere coverage used in radiation computations wrt ocean'
+          ! 'fractional atmosphere coverage used in radiation computations wrt ocean' (computed in med_phases_prep_ocn)
           call addfld(fldListTo(compocn)%flds, 'Sf_afracr')
-       end if
-    else
-       if (flds_i2o_per_cat) then
-          call addmap(fldListFr(compice)%flds, 'Si_ifrac_n', compocn, mapfcopy, 'unset', 'unset')
+          ! 'net shortwave radiation times atmosphere fraction' (computed in med_phases_prep_ocn)
+          call addfld(fldListTo(compocn)%flds, 'Foxx_swnet_afracr')
+       else
+          call addmap(fldListFr(compice)%flds, 'Si_ifrac_n'        , compocn, mapfcopy, 'unset', 'unset')
           call addmap(fldListFr(compice)%flds, 'Fioi_swpen_ifrac_n', compocn, mapfcopy, 'unset', 'unset')
-          ! TODO (mvertens, 2018-12-21): add mapping and merging
+          ! Note that 'Sf_afrac, 'Sf_afracr' and 'Foxx_swnet_afracr' will have explicit merging in med_phases_prep_ocn
        end if
     end if
 
     ! ---------------------------------------------------------------------
     !  to ocn: precipitation rate water equivalent from atm
     !  to ocn: snow rate water equivalent from atm
     ! ---------------------------------------------------------------------
+
     if (phase == 'advertise') then
-       do n = 1,size(iso)
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rainc'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rainl'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rain' //iso(n))
-          call addfld(fldListTo(compocn)%flds, 'Faxa_rain' //iso(n))
-
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snowc'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snowl'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snow' //iso(n))
-          call addfld(fldListTo(compocn)%flds, 'Faxa_snow' //iso(n))
-       end do
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainc')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainl')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rain' )
+       call addfld(fldListTo(compocn)%flds, 'Faxa_rain' )
```

Comment:
mixing of fldListTo and fldListFr seems confusing and potentially error prone.  
Can we have two loops here, one over fldlistFr and another over FldListTo?